lms/djangoapps/shoppingcart/models.py

```diff
@@ -836,6 +903,7 @@ class CourseRegistrationCode(models.Model):
     created_at = models.DateTimeField(default=datetime.now(pytz.utc))
     order = models.ForeignKey(Order, db_index=True, null=True, related_name="purchase_order")
     invoice = models.ForeignKey(Invoice, null=True)
+    invoice_item = models.ForeignKey(CourseRegistrationCodeInvoiceItem, null=True)
```

Comment:
Discussed with @aamir-khan and @awais786 in person: in the case of a rollback, we'd be running old code with the new data model.  Since the old code expects `invoice` to be populated, the new code needs to write to both fields.

In a future release, we can remove the `invoice` field and just use `invoice_item`.